src/game/stage/demo_manager.c

```diff
@@ -0,0 +1,217 @@
+    s->variant = gPressStartTiles[lang].variant;
+
+    s->unk14 = 0;
+    s->unk21 = 0xFF;
+    s->unk1C = 0;
+    s->unk22 = 0x10;
+    s->palId = 0;
+    s->unk1A = 0x40;
+    s->unk10 = 0;
+
+    if (blendCtrl != BLDCNT_EFFECT_BLEND) {
+        s->unk10 = SPRITE_FLAG(OBJ_MODE, ST_OAM_OBJ_BLEND);
+    }
+    sub_8004558(s);
+
+    s = &dm->textDemoPlay;
+    s->x = (DISPLAY_WIDTH / 2);
+    s->y = (DISPLAY_HEIGHT / 2);
+
+    s->graphics.dest = VramMalloc(28);
+    s->graphics.size = 0;
+    s->graphics.anim = SA2_ANIM_DEMO_PLAY;
+    s->variant = 0;
+
+    s->unk14 = 0;
+    s->unk21 = 0xFF;
+    s->unk1C = 0;
+    s->unk22 = 0x10;
+    s->palId = 0;
+    s->unk1A = 0x40;
+    s->unk10 = 0;
+
+    if (blendCtrl != BLDCNT_EFFECT_BLEND) {
+        s->unk10 = SPRITE_FLAG(OBJ_MODE, ST_OAM_OBJ_BLEND);
+    }
+    sub_8004558(s);
+}
+
+void Task_800A110(void)
+{
+    DemoManager *dm = TaskGetStructPtr(gCurTask);
+
+    if (gPhysicalInput & START_BUTTON) {
+        gPlayer.moveState |= MOVESTATE_IGNORE_INPUT;
+        gPlayer.unk5C = 0;
+
+        gUnknown_030054E4 = 1;
+
+        dm->playerPressedStart = TRUE;
+
+        gCurTask->main = Task_800A310;
+
+        gBldRegs.bldCnt = (BLDCNT_EFFECT_LIGHTEN | BLDCNT_TGT1_ALL);
+        gBldRegs.bldY = 0;
+
+        gUnknown_030054A8.unk0 = 0xFF;
+        CreateMusicFadeoutTask(64);
+    } else if (gCheckpointTime > (u32)ZONE_TIME_TO_INT(0, 24.5)) {
+        dm->playerPressedStart = FALSE;
+
+        gCurTask->main = Task_800A310;
+
+        gBldRegs.bldCnt = (BLDCNT_EFFECT_LIGHTEN | BLDCNT_TGT1_ALL);
+        gBldRegs.bldY = 0;
+
+        gUnknown_030054A8.unk0 = 0xFF;
+        CreateMusicFadeoutTask(64);
+    }
+
+    if (!(gUnknown_03005424 & EXTRA_STATE__100)) {
+        Sprite *s = &dm->textPressStart;
+
+        if (gUnknown_03005590 & 0x20) {
+            if (gBldRegs.bldY != 0) {
+                s->unk10 |= SPRITE_FLAG(OBJ_MODE, ST_OAM_OBJ_BLEND);
+            } else {
+                s->unk10 &= ~SPRITE_FLAG_MASK_OBJ_MODE;
+            }
+
+            sub_8004558(s);
+            sub_80051E8(s);
+        }
+
+        if (gBldRegs.bldY != 0) {
+            dm->textDemoPlay.unk10 |= SPRITE_FLAG(OBJ_MODE, ST_OAM_OBJ_BLEND);
+        } else {
+            dm->textDemoPlay.unk10 &= ~SPRITE_FLAG_MASK_OBJ_MODE;
+        }
+
+        sub_80051E8(&dm->textDemoPlay);
+    }
+}
+
+void Task_800A24C(void)
+{
+    DemoManager *dm = TaskGetStructPtr(gCurTask);
+    dm->unk62++;
+
+    m4aMPlayVolumeControl(&gMPlayInfo_BGM, 0xFFFF, 0);
+    m4aMPlayVolumeControl(&gMPlayInfo_SE1, 0xFFFF, 0);
+    m4aMPlayVolumeControl(&gMPlayInfo_SE2, 0xFFFF, 0);
+    m4aMPlayVolumeControl(&gMPlayInfo_SE3, 0xFFFF, 0);
+
+    if (dm->unk62 >= 0x30) {
+        gLoadedSaveGame->timeLimitDisabled = dm->timeLimitDisabled;
+        TasksDestroyAll();
+        gUnknown_03002AE4 = gUnknown_0300287C;
+        gUnknown_03005390 = 0;
+        gVramGraphicsCopyCursor = gVramGraphicsCopyQueueIndex;
```

Comment:
At some point need to create a macro for this. Like a reset

Reply:
Yup, thought that as well